----------------------------------------------------------------------
Patch name: Bochs Icon Problem Fix
Author: Volker Ruppert [Volker.Ruppert@t-online.de]
Date:

Detailed description:

	This patch fixes the failure on Win2000 related 
        to Bochs new icon problem.

Patch was created with:
  cvs diff -u
Apply patch to what version:
  cvs checked out on DATE, release version VER
Instructions:
  To patch, go to main bochs directory.
  Type "patch -p0 < THIS_PATCH_FILE".
----------------------------------------------------------------------
diff -urNX /home/volker/exclude-diffw --strip-trailing-cr 
/home/volker/bochs/gui/win32.cc ./gui/win32.cc
--- ../bochs/gui/win32.cc	2002-12-26 19:24:40.000000000 +0100
+++ ./gui/win32.cc	2002-12-29 09:40:24.000000000 +0100
@@ -376,12 +376,11 @@
 VOID UIThread(PVOID pvoid) {
   MSG msg;
   HDC hdc;
-  WNDCLASSEX wndclass;
+  WNDCLASS wndclass;
   RECT wndRect, wndRect2;
 
   workerThreadID = GetCurrentThreadId();
 
-  wndclass.cbSize = sizeof (wndclass);
   wndclass.style = CS_HREDRAW | CS_VREDRAW;
   wndclass.lpfnWndProc = mainWndProc;
   wndclass.cbClsExtra = 0;
@@ -393,9 +392,8 @@
   wndclass.lpszMenuName = NULL;
   wndclass.lpszClassName = szAppName;
 
-  RegisterClassEx (&wndclass);
+  RegisterClass (&wndclass);
 
-  wndclass.cbSize = sizeof (wndclass);
   wndclass.style = CS_HREDRAW | CS_VREDRAW;
   wndclass.lpfnWndProc = simWndProc;
   wndclass.cbClsExtra = 0;
@@ -407,7 +405,7 @@
   wndclass.lpszMenuName = NULL;
   wndclass.lpszClassName = "SIMWINDOW";
 
-  RegisterClassEx (&wndclass);
+  RegisterClass (&wndclass);
 
   stInfo.mainWnd = CreateWindow (szAppName,
                      szWindowName,
